dd-trace/src/configuration/configuration.rs

```diff
@@ -156,53 +181,56 @@ impl<T: Deref> Deref for ConfigItemRef<'_, T> {
     }
 }
 
+impl<T: std::fmt::Display> std::fmt::Display for ConfigItemRef<'_, T> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        (**self).fmt(f)
+    }
+}
+
+trait ConfigurationValueProvider {
```

Comment:
This needs documentation. The difference between `get_configuration_value` and `value` in `ConfigItem` is unclear from just reading the code.